tests

```diff
@@ -192,6 +192,29 @@ EOF
 		echo "              to only contain: [1-9]"
 	fi
 
+	testing $version password with secret:key syntax
+	./safe gen 64 secret/random:two
+	exitok $? 0
+	./safe get secret/random:two
+	exitok $? 0
```

Comment:
Can we get some sad-path tests on `safe gen`? There were a lot of new conditions to handle bad cases where we don't have the right patterns of `path key path key path:key path path:key key` type situations.